TRTexture16Importer/Helpers/ColourExtensions.cs

```diff
@@ -146,4 +148,60 @@ public static ushort ToRGB555(this Color c)
 
         return (ushort)(a | r | g | b);
     }
+
+    public static Color ToColor(this TRColour c)
+    {
+        int multiplier = TRConsts.Palette8Multiplier;
+        return Color.FromArgb(c.Red * multiplier, c.Green * multiplier, c.Blue * multiplier);
+    }
+
+    public static Color ToColor(this TRColour4 c)
+    {
+        return Color.FromArgb(c.Red, c.Green, c.Blue);
+    }
+
+    public static TRColour ToTRColour(this Color c)
+    {
+        int divisor = TRConsts.Palette8Multiplier;
+        return new()
+        {
+            Red = (byte)(c.R / divisor),
+            Green = (byte)(c.G / divisor),
+            Blue = (byte)(c.B / divisor)
+        };
+    }
+
+    public static TRColour4 ToTRColour4(this Color c)
+    {
+        return new()
+        {
+            Red = c.R,
+            Green = c.G,
+            Blue = c.B
+        };
+    }
+
+    public static int FindClosest(this List<Color> palette, Color colour, int startIndex = 0)
+    {
+        int colIndex = 0;
+        double bestMatch = double.MaxValue;
+
+        for (int i = startIndex; i < palette.Count; i++)
+        {
+            double match = Math.Sqrt
+            (
+                Math.Pow(colour.R - palette[i].R, 2) +
+                Math.Pow(colour.G - palette[i].G, 2) +
+                Math.Pow(colour.B - palette[i].B, 2)
+            );
+
+            if (match < bestMatch)
+            {
+                colIndex = i;
+                bestMatch = match;
+            }
+        }
+
+        return colIndex;
+    }
```

Comment:
Maybe we could change this to use more LINQ?

```C#
var indexed = palette.Select((value, index) => new {
    Index = index,
    Delta = Math.Sqrt(
        Math.Pow(colour.R - palette[i].R, 2) +
        Math.Pow(colour.G - palette[i].G, 2) +
        Math.Pow(colour.B - palette[i].B, 2)
    ),
});

return indexed.Min(item => item.Delta).Index;
```

Reply:
Thanks. Good point about `Math.Sqrt` - some unnecessary calculations there :) Updated now with LINQ - just had to ensure we skip over index 0 for TR1, so we honour the passed in start index.